src/zmalloc.c

```diff
@@ -715,6 +782,49 @@ int zmalloc_get_allocator_info(size_t *allocated, size_t *active, size_t *reside
     return 1;
 }
 
+/* Get the specified arena memory allocation information from allocator.
+ *
+ * For the meaning of the parameters, please refer to the function implementation
+ * and INFO's allocator_* in redis-doc. */
+int zmalloc_get_allocator_info_by_arena(unsigned int arena, size_t *allocated, size_t *active,
+                                        size_t *resident, size_t *frag_smallbins_bytes)
+{
+    char buf[100];
+    uint64_t epoch = 1;
+    size_t sz;
+    *allocated = *resident = *active = 0;
+
+    /* Update the statistics cached by mallctl. */
+    sz = sizeof(epoch);
+    je_mallctl("epoch", &epoch, &sz, &epoch, sz);
```

Comment:
maybe we can find a way to run the `epoch` only once per cron, and not twice?

Reply:
the only way i think of is that extrac a new updateEpoch function and then call it before the getallocatorinfo

Reply:
maybe check how severe is the impact (on a huge db), it think it is, and that means it is worthwhile to complicate the code for it.
maybe the safe approach is to add an argument to the function `refresh_stats`, so that the call to `updateEpoch` isn't forgotten.